modules/services/barrier.nix

```
@@ -0,0 +1,92 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let cfg = config.services.barrier;
+in {
+
+  meta.maintainers = with maintainers; [ kritnich ];
+
+  options.services.barrier = {
+
+    client = {
+
+      enable = mkEnableOption "Barrier Client daemon";
+
+      name = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Screen name of client. 
+          Defaults to hostname.
+        '';
+      };
+
+      server = mkOption {
+        type = types.str;
+        default = null;
+        description = ''
+          Server to connect to formatted as 
+          <literal>&lt;host&gt;[:&lt;port&gt;]</literal>.
+          Port defaults to 24800. Required.
+        '';
+      };
+
+      noTray = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Disables system tray icon.
+        '';
+      };
```

I think it would be cleaner to reverse the option and use `mkEnableOption`:

```suggestion
      tray = mkEnableOption "the system tray icon" // {
        default = true;
      };
```